V3.310 carbon fields

davideleuterius opened the pull request:
@designerbrent here is the actual branch I was on - my apologies!

davideleuterius committed:
#310 - adding carbon fields base files

davideleuterius committed:
#310 update for v3

designerbrent commented:
Ok, a few comments: 

- I pulled this branch and got a bit error. Turned out Carbon Fields was on included in this PR. I manually ran `composer require htmlburger/carbon-fields` and it pulled the necessary files down for version `3.1`. _Shouldn't this be included in the package?_
- After I had the updated theme, it didn't prompt for the upgrade as the v2 version does.
- I manually downloaded the `carbon-fields-converter` plugin and it worked, no errors!
- Piklist was still installed so all the settings and such.
- In the widgets, the Piklist AND Carbon Fields widgets are showing.

@davideleuterius It's looking good, I think.

davideleuterius commented:
@designerbrent 
It should be in the package, but my checking in my package would mean that the soberwp/controller package would be removed. This is part of the odd problem I was having with some of my local install environment being slightly off for reasons I still haven't figured out.

On this install you had, was this running version 3 already with Piklist? And had never had the converter plugin run before? There's an option that is inserted into the DB after running that plugin that prevents the upgrade screen from showing. If you hadn't done that, I'll have to see what that is happening then.

As far as Piklist being there, the script should absolutely remove all of that. Again that is something I will have to review closely by testing repeatedly. I am looking into these issues today.

designerbrent commented:
@davideleuterius I tried again to revert things back to what you checked in and removed the cf flag and the upgrade notice showed up. 

However, when it finished and I reloaded, I get the following error: 
```
 Fatal error: Uncaught Error: Class 'Carbon_Fields\Carbon_Fields' not found in /Users/hardingeb/Sites/alps/alps-wordpress-3/wordpress4/wp-content/themes/alps-wordpress-v3/app/fields.php:22 Stack trace: #0 /Users/hardingeb/Sites/alps/alps-wordpress-3/wordpress4/wp-includes/class-wp-hook.php(286): App\{closure}('') #1 /Users/hardingeb/Sites/alps/alps-wordpress-3/wordpress4/wp-includes/class-wp-hook.php(310): WP_Hook->apply_filters(NULL, Array) #2 /Users/hardingeb/Sites/alps/alps-wordpress-3/wordpress4/wp-includes/plugin.php(453): WP_Hook->do_action(Array) #3 /Users/hardingeb/Sites/alps/alps-wordpress-3/wordpress4/wp-settings.php(451): do_action('after_setup_the...') #4 /Users/hardingeb/Sites/alps/alps-wordpress-3/wordpress4/wp-config.php(89): require_once('/Users/hardinge...') #5 /Users/hardingeb/Sites/alps/alps-wordpress-3/wordpress4/wp-load.php(37): require_once('/Users/hardinge...') #6 /Users/hardingeb/Sites/alps/alps-wordpress-3/wordpress4/wp-admin/admin.php(31): require_once('/Users/hardinge...') #7 /Users/hardingeb/Sit in /Users/hardingeb/Sites/alps/alps-wordpress-3/wordpress4/wp-content/themes/alps-wordpress-v3/app/fields.php on line 22
```
I am running on Wordpress 4.x for this install.

davideleuterius commented:
@designerbrent Thanks for letting me know that. I'll see how I can adjust on my end to get this environment correct.

davideleuterius committed:
untrack composer vendor directory

davideleuterius committed:
adding composer file and fixing display bugs

- this will probably require running ‘composer update’

davideleuterius commented:
@designerbrent I have fixed an issue with composer that was causing a lot of trouble. You may need to run a composer update to get this running right. I am continuing to work on getting the install and conversion smooth and seamless.

designerbrent commented:
@davideleuterius That basically did the trick. I want to test it later this afternoon with a full top to bottom install, but it seemed to work in the quick test I did. I did notice that it is now saying it requires Piklist even through it removed it so probably just a flag that need to be checked.

davideleuterius commented:
@designerbrent That is great news. I am comparing results after conversion to make sure the data makes it. I've already found a couple of items to address but most of it seems to be converting successfully so far.

designerbrent commented:
@davideleuterius I just ran a second test and the conversion seems to work ok. 

However, I'm seeing that all the settings on the `posts` and `pages` are not working. Also some of the site-wide settings related to pages aren't working. Please check these over to ensure they are looking at the new CF settings.

davideleuterius commented:
@designerbrent I am working on getting everything ported over properly. Thanks for letting me know about what you found. I will be in touch with you later today and should have all of this resolved by then.

davideleuterius committed:
update for v3 data

- better handling of data for fields
- specifically posts template page
- improved conversion script

davideleuterius commented:
@designerbrent I believe this update should address most of your concerns. There are some items that I see that still have some issues, and I am wondering if any of these would be preexisting bugs? I am continuing to work on both the theme and the converter but this update should solve many issues. Please let me know what your experience is with this. The amount of issues that I have to fix appears to be something I can completely finish out by tomorrow unless you report anything from testing this new update. Thanks again for your patience!

designerbrent commented:
@davideleuterius Great to see this work progressing. I didn't do a deep test, but do see some improvements. There are some things below that I saw aren't working yet, but you probably are aware of them.

- [ ] Theme Settings: Copyright Statement doesn't show when there is no address.
- [ ] ALPS Gutenberg Blocks are aren't showing in the block editor but the plugin is install.
- [ ] The `post` settings don't work (dropcap, feature image layout, etc)
- [ ] The `page` settings are somewhat working. Related pages show but the images don't show. Heading settings don't work.
- [ ] Widgets show on the back end but don't render correct.

Happy to test more once you feel it warrants that.

davideleuterius commented:
@designerbrent Giving you a mid-day update on this. The following has been fixed:

In the admin for theme options:
- logo & footer logo now both convert AND can be edited / replaced. Was not saving properly before
- hide the sabbath column now converting and saving properly
- copyright is now showing independent of footer address

Additionally:
- hide the sabbath icon until scroll works in admin, but doesn't seem to work on the display side. Did this ever work properly?
- hide the sidebar is converting / saving in admin, but not sure how to test  this on the display side.
- under the posts tab, show images is working. This is where you toggle a site-wide option to show images wherever posts are called in a list. At least this is my understanding of how that works. 

This should completely cover and close out the theme options when we have these items resolved. The converter plugin is updated already and I am pushing some small changes for what has been done above for the V3 theme.

davideleuterius committed:
v3.310 update - theme options

davideleuterius committed:
adding CF for post feed round images

davideleuterius committed:
removing debug from footer

davideleuterius commented:
@designerbrent I am working on the post fields & display right now. I have a couple of questions regarding how things should display.

For posts, we can have the following:
- if a featured image has been set for the post (post thumbnail), then it will trigger the Featured Image Layout call in the sidebar column, which asks for a 1/3 2/3 or 2/3 1/3 layout for the title and featured image. The way this is currently set up, setting a featured image will always trigger the Featured Image Layout functionality.
- we then also have a 'hide featured image' checkbox. My thinking is that if you have checked this box, it will turn off the feature image display as expected, which would also seem to deactivate the Featured Image Layout functionality, returning a post to the default state for the header.
- we then also have the Template Addition fields, which includes a display title, a kicker, and a header background image. All of these show in the same area, but more like a banner. I would expect that these would be the lowest in the priority, so that if a featured image is set, the first item shows, but if the hide feature image is checked, then this would show if these fields were filled out.

Does that follow your thinking with this? Thanks!

davideleuterius commented:
For pages, we have a somewhat similar situation. Pages have the following:

- as always we have the featured image
- we have the Template Additions which has header background image, title, kicker, and hide featured image fields
- we have a Hero section for a banner with hero style, image, title, kicker, and link fields. It feels like if this Hero section is set, it should override everything else (Template Addition fields AND the featured image field) There can be three entries for this section.

It looks like the priority here should be this:

- Hero: if hero is set up, ignore Template Additions and Featured Image
- Template Additions: If hero is NOT set up, show if set up, and only if there is no featured image, or hide featured image has been checked
- Featured Image: show if set, do not show if Template Additions hide featured image has been checked, OR Hero has been set up

Does this sound correct to you? Thanks again!

davideleuterius committed:
post content type fixes

davideleuterius commented:
I have just pushed some fixes for the post content type:

- hiding dropcap works as expected
- all Template Addition fields now work as expected ( custom title, kicker, header bg image, hide sidebar, hide featured image )
- if the header background image is added in the Template Addition section, it will show even if the featured image is set, taking priority, and this will also show if the hide featured image box is checked 
( my thinking is the featured image is the post thumbnail in the WP sense. If this is incorrect, let me know )

designerbrent commented:
@davideleuterius Starting from the top:

[Theme Settings](https://github.com/adventistchurch/alps-wordpress/pull/356#issuecomment-535136033):
- Hide Until Scroll: I thought it was working but I can't seem to show proof. [You can see the demo of how it _should_ work here.](https://alps.adventist.io/v3/?p=pages-home-short)
- Hiding the sidebar should suppress all the `sidebar` widget areas on pages and posts.
- Show images: Should adjust things for the archive or category type pages. Does not affect the content at the bottom of `page` content.

[Post Settings](https://github.com/adventistchurch/alps-wordpress/pull/356#issuecomment-535157212):
- Featured image: not sure I fully understand. I'll have to see that in action.
- Hide image: Yes, that should hide it on the `post` page, not on the teaser view. [It should end up looking like this](https://alps.adventist.io/v3/?p=pages-longform)
- Template additions: I think we have it a little wrong in the current version. `pages` get the template additions and the ability to set a header image. But `posts` only have the page topper of the featured image or the version with no image. That should simplify things a bit.

[Page Settings](https://github.com/adventistchurch/alps-wordpress/pull/356#issuecomment-535162890)
- Hero / Template additions: Could we separate out the Template additions a bit? the `Hide the sidebar` and `Hide the Feature image` settings apply to all pages. However the `Display Title`, `Kicker` and `Header Background Image` options only apply when there is no `Hero`. Can those settings be combined somehow? 

Page image priority:
- Hero: if hero is set, ignore the page header and show the hero [Example](https://alps.adventist.io/v3/?p=pages-home)
- Template Additions: 'Header Background Image' shows _if_ no Hero is set.
- Feature Image: show as the background [example](https://alps.adventist.io/v3/?p=pages-basic-page-with-image) if it is not set to be hidden.
 
Does that make sense?

davideleuterius committed:
page content type fixes

davideleuterius committed:
added help text

davideleuterius commented:
@designerbrent I've just pushed fixes for the pages content type:

- Related Pages should be functional
  - one note on related pages, if you have round images checked, but display them in a grid, the images are not round. I think this is the desired functionality here though
- Template Additions should be functional now
- Hero fields should be functional now, and these take priority over the Template Additions and Featured Image in the same manner as I've outlined for posts.

designerbrent committed:
Fixing a spelling error in the help text.

designerbrent commented:
@davideleuterius in some quick testing on the `post` content type, the "ALPS Featured Image Hero Layout` setting does not change the image ratio. It only sits at 1/2 all the time.

davideleuterius commented:
> @davideleuterius in some quick testing on the `post` content type, the "ALPS Featured Image Hero Layout` setting does not change the image ratio. It only sits at 1/2 all the time.

@designerbrent Thank you for catching that. I'll get a fix in for that.

davideleuterius committed:
post hero layout fix

davideleuterius committed:
Merge branch 'v3.310-carbon-fields' of github.com:adventistchurch/alps-wordpress into v3.310-carbon-fields

davideleuterius commented:
@designerbrent - Got that fix in. Also about to respond to your earlier message now that I have a moment.

davideleuterius commented:
@designerbrent 
So repeating this back to you as I understand it:

**THEME / GLOBAL**
- Hide Until Scroll: the link is broken (that you linked to that is) but it looks like we just need to implement this functionality
- Hide Sidebar: hides any sidebar on home, any page, or any post, basically anywhere
- Show Images: toggles display of featured image with posts on any archive or category page

**POST**
- Hide Featured Image: only on the post itself
- Posts only have a featured image or no image at all, no option for a header background image

**PAGES**
- Hide Sidebar & Hide Featured Image should be separate options that are always shown
- If no Hero, then show the display title, kicker and header background image fields
- If Hero, do not show those fields at all

Also we are on the same page with the page image display priority. 

Thanks again for your patience. I think after these tweaks, this conversion part might be able to be considered complete.

designerbrent commented:
@davideleuterius sounds like we are on the same page on these items. Sorry about the missed url. Here is the correct one for `Hide Until Scroll`: https://alps.adventist.io/v3/?p=pages-home-short

designerbrent committed:
Fixing the classes on the author box from a double-nested look.

designerbrent committed:
Merge branch 'v3.310-carbon-fields' of github.com:adventistchurch/alps-wordpress into v3.310-carbon-fields

designerbrent commented:
There is an issue with the widgets still trying to load a template:
```
Warning: include(): Filename cannot be empty in /Users/hardingeb/Sites/alps/alps-wordpress-3/wordpress4/wp-content/themes/alps-wordpress-v3/app/carbon-fields/cf-widget.php on line 206

Warning: include(): Failed opening '' for inclusion (include_path='.:/Applications/MAMP/bin/php/php7.2.21/lib/php') in /Users/hardingeb/Sites/alps/alps-wordpress-3/wordpress4/wp-content/themes/alps-wordpress-v3/app/carbon-fields/cf-widget.php on line 206
```

davideleuterius commented:
@designerbrent Thanks for catching that. I'll be working on the widgets today to fix this issues.

davideleuterius committed:
fixing widget include path for v3

- this was pointing at a v2 location

designerbrent commented:
@davideleuterius Looking better but still showing what looks like the v2 block markup. 

![2019-09-26 at 1 14 PM](https://user-images.githubusercontent.com/132864/65709843-c699e480-e05f-11e9-9281-7027b16e6719.png)

Also: on the settings page for the `Post Feed` widget settings on the `Carbon Fields` version has a smaller set of options then the current `Piklist` version. Please double-check these options so we don't loose anything.

davideleuterius commented:
@designerbrent It seems this widget hasn't been coded up yet for version 3, so I am building that out this morning. I'll give you an update on general progress later today.

davideleuterius committed:
v3 changes for post feed widget

- new admin fields added or tweaked accordingly
- new display code & logic added

davideleuterius commented:
@designerbrent I now have the Post Feed Widget updated for V3. I see that the Text With Link Widget needs to be updated for the V3 styles as well, so I am working on that now.

davideleuterius committed:
V3 styling for Text With Link Widget

davideleuterius commented:
@designerbrent For a user, under Theme Settings, Post Options, Post Feed Image, if that is unchecked to prevent image display, does that also prevent the images from displaying on the page template Posts Template as well? Right now they are displaying, but would be simple enough to turn off if that is the expected behavior there. Thanks!

davideleuterius committed:
tweaks for theme, post and page lists

davideleuterius commented:
@designerbrent I've made tweaks to cover the points we discussed for theme, posts and pages. I am now getting to the urgent tickets to knock those out.

designerbrent commented:
On the widgets: 
The automatic inclusion of the line, padding and gray box needs to come out. I noticed it yesterday with the author box, and fixed it for just the author box, but it really should come out for everything and only get added back in as needed. [An example page](https://alps.adventist.io/v3/?p=templates-basic-page)
![2019-09-27 at 6 33 PM](https://user-images.githubusercontent.com/132864/65805974-9e8cad00-e155-11e9-83c1-27dbfb7ed247.png)

Images on post feed: 
It only affects the archive pages, not the post feed on pages or posts.

designerbrent commented:
@davideleuterius I think the rest looks good. I don't have time now, but I would like to do a full review of this one more time before saying it's all done. But I like where we are at! Great work.

designerbrent commented:
@davideleuterius Have you had a chance to work on the changes above? ☝️

designerbrent commented:
@davideleuterius I took another look at this and here are a few more specific notes on what is needed: 

1. This margin should be gone. There should be no margin around this block.
1. This margin is too wide.
1. The left side of this block should be against the left line, not inset like this.
1. The background on this block should be white.

![2019-10-01 at 1 11 PM](https://user-images.githubusercontent.com/132864/65984544-fffe9580-e44d-11e9-81db-9dc06f1f7e6e.png)

davideleuterius commented:
@designerbrent 
Thanks for the direction on this. I'll get these tweaks in for you now.

davideleuterius committed:
widget tweaks

- removing bg colors as directed
- adjusting spacing / padding as directed

davideleuterius commented:
@designerbrent 
I think these tweaks will bring the widgets up to speed for you. Let me know if they look ok for you now - thanks!

designerbrent commented:
@davideleuterius Closer but still off. 

- The search widget is spot on.
- The top widget (Text with Link) shouldn't have the rule or gray box under it.
- The Feed Widget should be lined up on the left edge.

![2019-10-01 at 5 16 PM](https://user-images.githubusercontent.com/132864/66001360-4fa18900-e46f-11e9-90bf-66b111824898.png)

davideleuterius commented:
@designerbrent This was actually fixed on my local system and still is, so I am trying to figure out why the fix didn't make its way to you. Really dumb question but can you do a hard refresh, just to be sure nothing is cached?

davideleuterius commented:
![screen](https://user-images.githubusercontent.com/4670424/66004653-f0dc0f80-e46e-11e9-9fe9-82284525e165.jpg)
This is what I see locally.

designerbrent commented:
@davideleuterius Are you sure everything got checked in? I did a pull again, I opened in a different browser, etc, an still not change. Any other places to look?

designerbrent commented:
@davideleuterius I tried again this AM by installing a new copy of Wordpress 5, installing the PikList version of the plugin, and then upgraded to the Carbon Fields version. Still getting the same results.

davideleuterius commented:
@designerbrent Thanks for letting me know. I'm working on getting the updated changes to you now.

davideleuterius committed:
widget sidebar spacing fixes

davideleuterius commented:
@designerbrent I think I have this fixed now - please take a look and let me know if you see the adjustments. Thanks!

davideleuterius committed:
#350 - shortcode fix

- added call for processing shortcode in the Text With Link widget

davideleuterius committed:
#350 - shortcodes in excerpts

- added filters for processing shortcodes in page / post excerpts

davideleuterius committed:
#350 - excerpts

- added to get_the_excerpt as well just in case

designerbrent commented:
@davideleuterius you got it! This fixed the issue. I will run through testing this again, but I think this works.

One thing I did note this morning on a quick test was that when I upgraded, the widgets didn't move across to the new CF widgets.

davideleuterius commented:
@designerbrent Good news on the tweaks. I will look at the converter again. We've made some changes and I'll need to make sure we are in sync with everything.

designerbrent commented:
@davideleuterius whatever change was made for the excerpts is not removing the shortcodes and it is putting the teaser text from the current page into all the blocks.

![2019-10-02 at 10 42 AM](https://user-images.githubusercontent.com/132864/66054071-75776e00-e501-11e9-83ea-83fefc7a7c9b.png)

(Sorry to add this here but since you pushed the changes into this branch, I figured I should check it here.)

davideleuterius committed:
#350 - filtering for shortcodes

- adds shortcode filtering for related pages and post template components

davideleuterius commented:
@designerbrent I have just pushed up an adjustment for this that should fix this. This change is a little more atomic instead of global so this should be focused enough to avoid that unintended duplication.

davideleuterius committed:
#339 - new long form page header

- preserved old fields from previous page header

designerbrent commented:
@davideleuterius 
The header looks good 👍 

The related page has cleared the shortcodes 👍 

But the related pages still doesn't show the right content. It is showing the teaser for the current page on all related pages (same content for reach one.)

davideleuterius commented:
@designerbrent - OK - no problem. It seems it isn't pulling it with the loop. Should be an easy fix. Also I will revisit the converter plugin to make sure it is converting the widgets all the way through properly. I will check in with you early tomorrow morning with an update on both of these items.

davideleuterius commented:
@designerbrent I cannot seem to replicate your issue. Can you please tell me exactly what you have set up to get it to display this problem?

designerbrent commented:
Ok. Let me try and detail this: 

### With Child pages:

If you use the **Related Pages** _child pages_ setting, it returns the correct information but includes shortcodes and the `…` for images near the top.

![2019-10-03 at 4 48 PM](https://user-images.githubusercontent.com/132864/66163238-d84e2f80-e5fd-11e9-9472-c33a29521c6c.png)

### With the Custom page:

If you use the **Related Pages** _custom page_ setting, it shows the same text replicated for all the custom pages selected. 

![2019-10-03 at 4 50 PM](https://user-images.githubusercontent.com/132864/66163443-5b6f8580-e5fe-11e9-8a26-278fad34df99.png)

designerbrent commented:
@davideleuterius We seem to be very close on this. Hopefully the comment above clears up the problem I am seeing in the loop. ☝️

davideleuterius commented:
@designerbrent 
I have spent some time testing this and this appears to be functioning exactly as you want. I have attached screenshots that show a related pages with children / grandchildren with and without images, and a custom pages set up with and without images. Anywhere you see Hello world! in the text below is coming from a [test] shortcode that simply outputs the Hello world! text. 

I am not experiencing the duplicated content issue you are having so please let me know if that is continuing for you with this. I included these screens so that you could see that the content is different from item to item. 

![custom-no-img](https://user-images.githubusercontent.com/4670424/66513799-809e4100-eaa1-11e9-97d5-61c250af8b1f.jpg)
![custom-imgs](https://user-images.githubusercontent.com/4670424/66513800-809e4100-eaa1-11e9-8852-9114e9180d0e.jpg)
![children-imgs](https://user-images.githubusercontent.com/4670424/66513880-a88da480-eaa1-11e9-9afa-0cf5e166daab.jpg)
![children-no-img](https://user-images.githubusercontent.com/4670424/66513881-a88da480-eaa1-11e9-82a7-c7ab9ac703a2.jpg)

davideleuterius committed:
shortcode / duplicated loops issues

designerbrent reviewed (changes requested):
@davideleuterius I think this does it!

I only have one thing left: We need this to be setup so that it doesn't require the end user to run `composer install` as it does now. once that is done, I'm ready to go.

davideleuterius commented:
@designerbrent That sounds good. I am going to get that setup and also review the conversion plugin to make sure the widgets make it over properly.

davideleuterius committed:
adding vendor directory

- removes need for user to run -composer install-

davideleuterius committed:
updating gitignore

designerbrent committed:
Fixing the page header blade to remove the top margin.

designerbrent commented:
@davideleuterius in testing today, I just noticed that after it runs the update (and removes Piklist) it is still requiring Piklist.

Also, it hosed my menus.

davideleuterius commented:
@designerbrent I'm going back and readjusting the widgets and menus because we've got different widgets, different widget field names, and the menus may have changed as well. I finally have the theme detection working between v2 and v3, so I can now actually get these differences ironed out. Continuing to work on this to bring this to completion.

designerbrent commented:
I also noticed that there is a section for the "Long Header". I thought we talked about this before, but this should not be there. ALL the page headers should using this long header.

![2019-10-11 at 11 11 AM](https://user-images.githubusercontent.com/132864/66663177-32f81480-ec18-11e9-9c6d-be3d9d662050.png)

davideleuterius commented:
@designerbrent No problem on this - will change those labels.

designerbrent commented:
@davideleuterius I realized I might not have been clear on what I commented above, so let me try again: 

For the Wordpress theme, the ONLY header that will be used in the `Page Header - Long`. All titles will fall under that by default. We should have the option to overwrite it, as we did before, but by default, that will be the code used.

davideleuterius committed:
page header

davideleuterius committed:
Merge branch 'v3.310-carbon-fields' of github.com:adventistchurch/alps-wordpress into v3.310-carbon-fields

davideleuterius commented:
@designerbrent 
I've updated these fields. Did I need to remove the Hero options for pages, as that will override the page header?

davideleuterius commented:
@designerbrent 
The converter plugin is not actually touching the menus at all, so I think what is happening with that is that we have differently named menus or menu areas from V2 to V3.

I have set up a couple of menus in V2 and then set / converted the theme to V3. Below are some screenshots that show what I am talking about. First is V2:


<img width="1346" alt="v2-edit-menus" src="https://user-images.githubusercontent.com/4670424/66857315-a4073700-ef4c-11e9-8fb7-4d685ff7457f.png">
<img width="781" alt="v2-edit-menus-2" src="https://user-images.githubusercontent.com/4670424/66857330-a8335480-ef4c-11e9-8374-cc267574f309.png">
<img width="1200" alt="v2-menu-display" src="https://user-images.githubusercontent.com/4670424/66857334-ad909f00-ef4c-11e9-9d65-2a5e70e7b9c9.png">

**This is after setting / converting to V3:**

<img width="1206" alt="v3-edit-menus" src="https://user-images.githubusercontent.com/4670424/66857356-bbdebb00-ef4c-11e9-9f05-08343c195e69.png">
<img width="1210" alt="v3-edit-menus-2" src="https://user-images.githubusercontent.com/4670424/66857370-c305c900-ef4c-11e9-82f1-d17df140c6e6.png">
<img width="880" alt="v3-menu-display" src="https://user-images.githubusercontent.com/4670424/66857384-c8631380-ef4c-11e9-9ddb-4b572f77c1ee.png">

So I think this may be an issue of the menu name and menu location being different between the two versions? Unless there is something else happening with these menus?

This may be a matter of making sure the menu is assigned to the right theme location?

davideleuterius commented:
So this is where this is being set:

V2

` register_nav_menus([
    'primary_navigation' => __('Primary Navigation', 'sage'),
    'secondary_navigation' => __('Secondary Navigation', 'sage'),
    'footer_secondary_navigation' => __('Secondary Footer Navigation', 'sage'),
    'footer_navigation' => __('Primary Footer Navigation', 'sage')
  ]);`

V3
`
   register_nav_menus([
      'primary_navigation' => __('Primary Navigation', 'alps'),
      'secondary_navigation' => __('Secondary Navigation', 'alps'),
      'learn_more_navigation' => __('Learn More', 'alps'),
      'footer_primary_navigation' => __('Footer Primary Navigation', 'alps'),
      'footer_secondary_navigation' => __('Footer Secondary Navigation', 'alps')
    ]);`

So we have a new menu and location ( learn_more_navigation, Learn More ) and the footer navigation is named differently. Should I change the location name to be consistent from V2 to V3? I think the one that would be breaking here would be the main footer navigation (V2: footer_navigation vs V3: footer_primary_navigation )? The label name wouldn't cause it to break, but renaming the theme location for the menu definitely will. However this will break that menu for any current V3 user.

The other option to avoid breaking it for current V3 users would be to tell users to make sure their menus are set for the proper areas. Let me know how you want to set that up - thanks!

designerbrent commented:
@davideleuterius no, Hero's can stay.

designerbrent commented:
@davideleuterius With regards to the menus, the switch I saw was not between `v2` and `v3`, it was between `v3` and `v3` with Carbon Fields. That said, I **don't feel** it is worth holding up release on this. Every site that is running this now is more or less beta.

davideleuterius commented:
@designerbrent OK, got it on the menus. I am still working on the widget issue right now. I hope to have that ironed out by the end of today.

davideleuterius commented:
@designerbrent After a lot of testing, I have finally figured out why widgets are not being converted from V2 to V3. The problem is that to convert to V3, we have to SET the theme to V3. Once we do that, WordPress switches over to a completely different set of widgets, leaving anything having to do with V2 widgets behind. While menus are not theme independent, widgets are very much theme dependent. I am going to see if I can find a way to work around this issue.

EDIT: This would happen even just going from V2 to V3, or really from any theme to any other theme. However if updating to Carbon Fields on V2 this would work fine. The actual issue here is changing themes. I could however read the previously set widgets and manually assign them to new areas, like setting the Post Feed widget to being in the Post Template Sidebar widget area for example.

davideleuterius commented:
@designerbrent 
I am going to go ahead and read for any post feed widgets when going from V2 to V3 and manually assign them to the Posts Template Sidebar widget area. Once this is done, this should complete 310/356?

designerbrent commented:
@davideleuterius Yes, i think that should do it.

designerbrent commented:
@davideleuterius I just had a chance to pull the latest release and test it. The header isn't quite what I was thinking. Let me try here:

- The `Carbon Fields` section on the admin page looks great. It allows an admin to override some aspects of the page.
- When it comes to the template, `page-header.blade.php`, that is what needs to be changed. In that file, [lines 63-80](https://github.com/adventistchurch/alps-wordpress/blob/v3.310-carbon-fields/views/patterns/02-organisms/sections/page-header.blade.php#L63) (approximately) need to come out.
- The ONLY header to be used on all the pages is the long version below:
![2019-10-16 at 2 30 PM](https://user-images.githubusercontent.com/132864/66947838-98397480-f021-11e9-8017-c44c20b797cb.png)

davideleuterius commented:
@designerbrent Thank you so much for clarifying that. I will make that change now.

davideleuterius committed:
removing piklist as required, header cleanup

davideleuterius commented:
@designerbrent A quick update for you: because ALL of the widgets from V2 -> V3 don't convert in place, I thought it would be best to convert them and then place them in the Inactive Widgets area. Right now the post feed widget is converting this way, and will have to do it for the other ones, which now that I have that working, shouldn't take too long by comparison. I'll need to double check it again from setting them up from the V2 theme with Piklist and then switching to V3 and running the converter, which is how I would recommend theme users to migrate if they are on V2. I'll let you know as soon as I have it handling it orphaned V2 widgets and placing them all in inactive widgets. From there a user can drag them to the widget area they want at that point.

designerbrent commented:
@davideleuterius 
Glad to hear progress is happening. While I would love to have all the widgets move, I think it's not too much for people to re-look at their widgets after a theme upgrade like this.

On the headers: As it currently sits, when there is no content in the `Header Banner` CF area, **NO** banner shows. As I said before above, the banner should be the same on every page, no matter if the `Header Banner` has content or not, pulling from the main `Title` area, the typical kicker content of the parent page, etc.

davideleuterius committed:
making ‘long’ header global

davideleuterius committed:
limit banner header to only pages

davideleuterius committed:
page header

- catching unset variable

davideleuterius committed:
handling home, archive & post template header

- sending title far left on these

davideleuterius committed:
fixing excerpts

davideleuterius commented:
@designerbrent 
I've made some changes for the page header. I think I have this working as you have directed. If I recall correctly, the new page header was only for pages and not posts? If not, it is easy enough to set globally.

Also I think I have the converter plugin now handling the widgets and putting them in the inactive widgets area if moving from V2 to V3. It worked for me just now with my tests so if you could give it a try and let me know that would be great! Thank you so much!

designerbrent commented:
@davideleuterius I'm happy with the design and function of the theme now. 👍 

However, I'm trying to test an upgrade from `v2` to `v3` and running into issues. Here is what I did:

1. Installed the v2 theme.
1. Added some custom widgets
1. Switched to the `v3` theme (`v3.310-carbon-fields`). 
1. The theme switch required some additional modules. However, when I hit the **Install Required Plugins** page, it gave me several warnings: 
![2019-10-22 at 7 09 AM](https://user-images.githubusercontent.com/132864/67282105-f1c8f580-f49e-11e9-92a2-f33e9227015d.png)
1. After this, it still said that the `alps-gutenberg-blocks` plugin was not installed. Trying to install it again, it ended up nested in another folder, making Wordpress think it's installed: 
![2019-10-22 at 7 43 AM](https://user-images.githubusercontent.com/132864/67282420-c85c9980-f49f-11e9-80de-563407eb5b6f.png)
1. Additionally, in Appearance, there are two theme settings pages: 
![2019-10-22 at 7 45 AM](https://user-images.githubusercontent.com/132864/67282619-330dd500-f4a0-11e9-971f-fb13c9e90675.png)
1. `Piklist` is still required.

Any thoughts on getting through this?

davideleuterius commented:
@designerbrent 
Thanks for checking in with those issues.

Those warnings look like they are coming from the internal workings of WordPress itself, but could be from the way that the required plugins are trying to call them. I will look into how we can resolve those errors.

Also, that is really odd with the nested folder issue. However that may be a result of the above issue. 

When you switched to V3, did you run the converter? Because until you run the converter, you'll temporarily have that kind of set up. The only other thing I could do there is that on activating V3 I can try to have it run the converter plugin.

davideleuterius commented:
@designerbrent 
It looks like something changed with the way the ALPS Gutenberg Blocks is being delivered to WP. For some reason, it is being 'double wrapped' in a directory. You can verify this by downloading this file, which is what WP is pulling down for the ALPS Gutenberg Blocks plugin:

https://kernl.us/api/v1/updates/5c13a3859e9cea4aa2fd8fbd/download

When you extract this, you should see it 'double wrapped' as I've described. I've successfully downloaded this in the past without this issue, so something somewhere must have changed. You can see that the source repo for pulling the plugin has not changed structure:

https://github.com/adventistchurch/alps-gutenberg-blocks

WP will not install a plugin that doesn't have a valid plugin php file in the root directory, so it is looking at what's been extracted and saying there is nothing there to install, so it doesn't actually install at all, even though the files are there, albeit in the wrong location.

I tried installing Guideposts manually by itself and it installed without any issues. 

Also at this point Piklist should not be required at all. I had commented it out from functions.php, but I am now going to completely remove it in the code. I will push this up shortly.

davideleuterius committed:
removing Piklist requirement from code

designerbrent commented:
@davideleuterius So it seems there was an issue with Kernl or something but I got that to download correctly now, so we are good with that.

I did get an error when I walked through the upgrade process again today. It showed my the ALPS update screen (The large, obnoxious warning to upgrade) but as the converter ran, it said `The plugin generated unexpected output.`. After going back to the site, I found that it was still trying to use some Piklist widgets, Piklist was still locked on so it wouldn't uninstall, and there were some other errors on the page in the widget areas: 
![2019-10-23 at 4 43 PM](https://user-images.githubusercontent.com/132864/67432432-427e4280-f5b4-11e9-97f0-26c2cee8b18f.png)

Also: there are now some conflicts on the branch that need to be resolved.

designerbrent commented:
@davideleuterius I just ran a test on porting over a v2 site to this branch of v3 and it went very well. Missed a couple minor things (menus) but nothing I want to worry about for this.

There are a few concerns that I have and once those are resolved, I think we can merge this into the main `v3` branch: 

- [x] On the `Related Pages` area, can we change that to `Related Pages & Posts` as it currently functions in `v2`? In my testing, it is only showing pages, though it labels all of them as posts. The example below should have 34 items but it only has 17 and they are all posts.
![2019-10-24 at 12 02 PM](https://user-images.githubusercontent.com/132864/67503776-3d71cf80-f656-11e9-9dc3-3478326ea04a.png)
- [ ] Is there a functional reason why the ALPS Carbon Fields are highlighted in gray and larger then the others?
- [ ] I got all the way through the upgrade from `v2` to `v3` and found that the site was still on `Wordpress 4.x`. Can we make `Wordpress 5.x` a requirement for `v3` and the `ALPS Gutenberg Blocks`? 
- [ ] Please fix the file conflicts so the branch is ready to merge.

davideleuterius commented:
@designerbrent 

Addressing your four items in your last post:

- 1 - Related Pages - no problem, I can make this pull in both pages and posts.
- 2 - There is no functional reason, other than at first it was helping me find those fields in the admin. The styling for that can be removed if you would rather not have that display that way.
- 3 - I will need to see how we can force users to upgrade to WP V5 *before* the actual conversion takes place.
- 4 - I will resolve these conflicts as well.

I will check in with you with a progress update a little later today.

davideleuterius committed:
related posts & pages

davideleuterius committed:
CF admin fields styling removal

davideleuterius committed:
CF admin styling

davideleuterius committed:
direct users to upgrade WP

davideleuterius commented:
@designerbrent I've got 3 of the 4 items addressed so far. Working on the conflicts now, and I will then look at that last Piklist widget error you're describing above.

davideleuterius commented:
@designerbrent 
On item #1, Carbon Fields is labelling them all as "posts" regardless if they are posts or pages. I'm not sure how to fix this as it considers pages as posts (which is technically true, but ... ) in how it maps out content types.

designerbrent commented:
@davideleuterius I can live with Carbon Fields labeling everything as posts. I don’t want you reinventing that.

davideleuterius committed:
Merge branch 'resolve-conflicts-v3' into v3.310-carbon-fields

# Conflicts:
#	dist/scripts/customizer.js.map
#	dist/scripts/main.js.map
#	dist/styles/main.css
#	dist/styles/main.css.map

davideleuterius commented:
@designerbrent I believe I have that last issue ironed out for you with the conversion, and I just resolved the conflicts. Let me know if there is anything else I can do.

davideleuterius committed:
important: correcting theme name

designerbrent committed:
Adjustments the the behavior of the large header.

designerbrent committed:
Fixing an issue with extra spacing on the hero image.

designerbrent committed:
Removing Piklist and replacing it with Carbon Fields.
